website/docs/recipes/munit.md

```diff
@@ -0,0 +1,115 @@
+---
+id: munit
+title: MUnit (Mule4) with Pact
+description: How to use Pact for testing MuleSoft (Mule 4) APIs with MUnit
+---
+
+# Problem Statement: Pact with Mule 4 (MuleSoft)
+
+As of Mule 4, it is no longer possible to extend the MUnit runner through Java code. Previously, the solution of Pact testing with MuleSoft was to extend “FunctionalMunitSuite” in your test class. An example can be found here: https://docs.mulesoft.com/munit/1.3/munit-tests-with-Java#creating-your-suite-class
+
+However, MuleSoft now wants alignment with the Mule language to be able to provide a better service. Which leaves the outstanding problem; How can we use Pact to test Mule 4 applications? 
+
+In a nutshell, the solution is to have MUnit be the test runner and execute the standalone pact-JVM server by calling Java code within the MUnit console. This will allow for both the benefit of MUnit tests along with pact files being created for contract testing. As a result, we will be able to leverage the MUnit unit test coverage while also generating our pact files and publishing them to a Pact Broker or sharing them into your own file storage system. 
+
+# Consumer
+
+## So how do we do it?
+
+First, we need to understand the MUnit test layout. It has a very similar test structure to other unit testing tools, such as Jest, jUnit, nUnit, etc., with the usual before suite, before test, after test, and after suite. The test logic is formed into the test blocks where “flows” are executed in the test scripts. An empty MUnit test would look like the following:
+
+ ![Anypoint Studio MUnit Test Layout](../../static/img/docs/recipes/munit/anypoint-test-struct.png)
+
+In the test block, we can see it is divided into three sections; execution, behavior, and validation. The assertions for tests come from MUnit tools providing a low code UI block that we configure for each type of unit test assertion we wish to perform on the flow. 
+
+Looking through the Mule Palette, in addition to these low code blocks there are also Java blocks for writing Java classes with the ability to call functions. The functionality of these blocks is as follows:
+•	New: creates a new class
+•	Invoke static: invoke static methods
+•	Invoke: invoke methods on an instantiated class
+•	Validate type: check instance is of a given class
+
+ 
+![Mule Java Palette](../../static/img/docs/recipes/munit/mule-palette-java.png)
+
+For this recipe, we will be utilizing all but the ‘Validate type’ Java block, but you are welcome to use it in your own solution. 
+
+## Starting the Pact JVM from Java
+
+To use the standalone JVM of Pact we will need to create a static class called PactService. This will handle the command to run and close the Pact Standalone Server. The class will have the following functions:
+- startPactService: will start the service through the CLI command
+- stopPactService: will stop the Java process
+
+```Java
```

Comment:
this needs to be `java` to get nice pretty formatting

https://github.com/pact-foundation/docs.pact.io/blob/c69eada474479f6f0f1b6a42c6ae865763f38285/website/docusaurus.config.js#L203